gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcExceptionServerStreamingCallable.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.ResponseObserver;
+import com.google.api.gax.rpc.ServerStreamingCallable;
+import com.google.api.gax.rpc.StatusCode.Code;
+import com.google.api.gax.rpc.StreamController;
+import java.util.Set;
+import java.util.concurrent.CancellationException;
+
+/**
+ * Transforms all {@code Throwable}s thrown during a call into an instance of {@link ApiException}.
+ *
+ * <p>Package-private for internal use.
+ */
+class GrpcExceptionServerStreamingCallable<RequestT, ResponseT>
+    extends ServerStreamingCallable<RequestT, ResponseT> {
+  private final ServerStreamingCallable<RequestT, ResponseT> inner;
+  private final GrpcApiExceptionFactory exceptionFactory;
+
+  public GrpcExceptionServerStreamingCallable(
+      ServerStreamingCallable<RequestT, ResponseT> inner, Set<Code> retryableCodes) {
+    this.inner = inner;
+    this.exceptionFactory = new GrpcApiExceptionFactory(retryableCodes);
+  }
+
+  @Override
+  public void call(
+      RequestT request, ResponseObserver<ResponseT> responseObserver, ApiCallContext context) {
+
+    inner.call(request, new ExceptionResponseObserver(responseObserver), context);
+  }
+
+  private class ExceptionResponseObserver implements ResponseObserver<ResponseT> {
+    private ResponseObserver<ResponseT> innerObserver;
```

Comment:
I'm sorry; this is probably a really dumb question.
We have `ResponseObserver` and `ApiStreamObserver`. I understand why we need `ResponseObserver` but why is it that only server streaming call uses `ResponseObserver`? Shouldn't bidi use it too? It currently uses `ApiStreamObserver`.

I'm probably missing something.

Reply:
I wanted to scope my changes to features that can be used/tested in the bigtable client. This PR is special because it’s a user visible behavioral change for serverstreamingcallables and I wanted to make the breaking change at consistent across the api

Reply:
I see it now. Sorry I misread the code earlier.

Reply:
Gax, this pr changes the exceptions being thrown in all gax streaming callables

Reply:
Gax, this pr changes the exceptions being thrown in all gax streaming callables